test/amoc_api_scenario_handler_SUITE.erl

```diff
@@ -4,260 +4,149 @@
-    [{Answer,List}] = jsx:decode(JSON),
+    {CodeHttp, _Body} = patch_request(URL, RequestBody),
     %% then
-    ?assertEqual(200, CodeHttp),
-    ?assertEqual(<<"nodes">>, Answer),
-    ?assertEqual(true, is_list(List)).
+    %% Maybe check Body, as answer format will be ready
+    ?assertEqual(400, CodeHttp).
 
 
-test_get_tests_results(_) ->
+patch_scenario_returns_200_when_request_ok_and_module_exists(_Config) ->
     %% given
-    meck:new(amoc_test_event, [unstick, passthrough]),
-    TestResFun = fun(_, _) -> {ok, [{foo,test_begin}], [{foo, test_begin}]} end,
-    meck:expect(amoc_test_event, handle_call, TestResFun),
-    given_applications_started(),
-    %% when
-    Result = httpc:request("http://localhost:4000/test_status"),
-    {ok, {{_HttpVsn, CodeHttp, _Status}, _, Body}} = Result,
-    meck:unload(amoc_test_event),
-    BodyToParse = case erlang:is_bitstring(Body) of
-                    true ->
-                        Body;
-                    false ->
-                        erlang:list_to_bitstring(Body)
-                  end, 
-    [{Key, Value}] = jsx:decode(BodyToParse),
-    %% then
-    ?assertEqual(test_begin, erlang:binary_to_existing_atom(Value, utf8)),
-    ?assertEqual(foo, erlang:binary_to_existing_atom(Key, utf8)),
-    ?assertEqual(200,CodeHttp).
+    URL = get_url() ++ "/scenarios/sample_test",
+    RequestBody = jsx:encode([{users,10}]),
 
-test_get_wrong_test_result(_) ->
-    %% given
-    given_applications_started(),
-    %% when
-    Result = httpc:request(post,{"http://localhost:4000/test_status", [], "application/json", "{\"name\": \"foo\"}"}, [], []),
-    {ok, {{_HttpVsn, CodeHttp, _Status}, _, _}} = Result,
-    %% then
-    ?assertEqual(500, CodeHttp).
-    
-test_get_test_result(_) ->
-    %% given
-    meck:new(amoc_test_event, [unstick, passthrough]),
-    TestResFun = fun(_, _) -> {ok, test_crashed, [{foo, test_crashed}]} end,
-    meck:expect(amoc_test_event, handle_call, TestResFun),
-    given_applications_started(),
+    meck:new(amoc_dist, [unstick]),
+    Fun = fun(_,1,_) -> ok end,
+    meck:expect(amoc_dist, do, Fun),
     %% when
-    Result = httpc:request(post,{"http://localhost:4000/test_status", [], "application/json", "{\"name\": \"foo\"}"}, [], []),
-    {ok, {{_HttpVsn, CodeHttp, _Status}, _, Body}} = Result,
-    meck:unload(amoc_test_event), 
-    BodyToParse = case erlang:is_bitstring(Body) of
-                    true ->
-                        Body;
-                    false ->
-                        erlang:list_to_bitstring(Body)
-                  end,
-    [{Key, Value}] = jsx:decode(BodyToParse),
+    {CodeHttp, _Body} = patch_request(URL, RequestBody),
     %% then
-    ?assertEqual(test_crashed, erlang:binary_to_existing_atom(Value, utf8)),
-    ?assertEqual(foo, erlang:binary_to_existing_atom(Key, utf8)),
+    %% Maybe check Body, as answer format will be ready
+    meck:wait(amoc_dist, do, ['sample_test', 1, 10], 1000),
+    meck:unload(amoc_dist),
```

Comment:
We could even extract loading/unload the mocked module into the init_per_testcase.